src/components/sidebar/main-sidebar.tsx

```diff
@@ -149,6 +140,20 @@ export const MainSidebar: React.FC<MainSidebarProps> = ({ user, setShowDashboard
   const renderDashboardSetupModal = () => {
     const portalElement = document.getElementById('dashboard-setup-portal');
     if (!portalElement) return null;
+    useEffect(() => {
```

Comment:
you have the same use effect hook twice in the same page dont know why

Reply:
143-156 and 187-199

Reply:
on top of that, I cant approve this method because it doesnt restore the original functionality. The page still reloads and this adds unnecessary complexity to parse the pathname again on each reload. Instead restore how it used to work before?